tools/tsp-client/src/typespec.ts

```diff
@@ -0,0 +1,154 @@
+import { NodeHost, compile, getSourceLocation } from "@typespec/compiler";
+import { parse, isImportStatement } from "@typespec/compiler";
+import { Logger } from "./log.js";
+import { readFile, readdir } from "fs/promises";
+import * as path from "node:path";
+import { parse as parseYaml } from "yaml";
+
+
+export async function getEmitterOptions(rootUrl: string, tempRoot: string, emitter: string, saveInputs: boolean, additionalOptions?: string): Promise<Record<string, any>> {
+  // TODO: Add a way to specify emitter options like Language-Settings.ps1, could be a languageSettings.ts file
+  let emitterOptions: Record<string, Record<string, unknown>> = {};
+  emitterOptions[emitter] = {};
+  if (additionalOptions) {
+    emitterOptions = resolveCliOptions(additionalOptions.split(","));
+  } else {
+    const configData = await readFile(path.join(tempRoot, "tspconfig.yaml"), "utf8");
+    const configYaml = parseYaml(configData);
+  
+    if (configYaml["options"] && configYaml["options"][emitter]){
+      emitterOptions[emitter] = configYaml["options"][emitter];
+      for (const key in emitterOptions[emitter]!) {
```

Comment:
for/of Object.entries would likely suit you much better

Reply:
This code was changed.